com.unity.visualeffectgraph/Editor/FilterPopup/VFXFilterWindow.cs

```diff
@@ -660,36 +634,20 @@ private void ListGUI(Element[] tree, GroupElement parent)
             }
         }
 
-        private List<Element> GetChildren(Element[] tree, Element parent)
+        private IEnumerable<Element> GetChildren(Element[] tree, Element parent)
         {
-            List<Element> children = new List<Element>();
-            int level = -1;
-            int i = 0;
-            for (i = 0; i < tree.Length; i++)
-            {
-                if (tree[i] == parent)
-                {
-                    level = parent.level + 1;
-                    i++;
-                    break;
-                }
-            }
-            if (level == -1)
-                return children;
-
-            for (; i < tree.Length; i++)
-            {
-                Element e = tree[i];
-
-                if (e.level < level)
-                    break;
-                if (e.level > level && !hasSearch)
-                    continue;
-
-                children.Add(e);
-            }
+            var childrenLevel = parent.level + 1;
+            return tree.SkipWhile(x => x != parent)
+                .Skip(1)
+                .TakeWhile(x => x.level >= childrenLevel)
+                .Where(x => x.level == childrenLevel || (x.level > childrenLevel && hasSearch));
+        }
```

Comment:
I prefer the Linq approach than the previous difficult to read algorithm, but you have to be careful modifying this : Linq doesn't perform as fast as general .NET application on Unity. However, in your case, it seems correct.